### PR TITLE
Add issue/PR templates, CoC, CONTRIBUTING.md

### LINKED ISSUE #1: Add issue/PR templates
#### Description
Add the issue and PR templates so we can have a centralized definition of those.

### LINKED ISSUE #3: Add CoC/contributing.md
CoCs and contributing.mds are currently located in `docs/` folders in each repo. Should we centralize those in `.github` as well?

### PR DESCRIPTION
**Pull request by chriszs**
## Description
This PR adds the issue and pull request templates and docs so we can have a centralized definition of these.

## Related Issues
Resolves #1 
Resolves #3

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{ This PR fixes ... }
+
+{ This PR removes ... }
+
+## Due Diligence Checklist
+- [ ] Tests
+- [ ] Documentation
+- [ ] Consolidated commits
+- [ ] Relevant labels assigned
+- [ ] Changelog(s) updated
```
Comment (chriszs) [inline pull_request_template.md:13]: `Changelog(s) updated` is the only part of this template that seems like it should vary depending on the repo. Should I leave this out?

Comment (slifty) [reply pull_request_template.md:13]: I wouldn't mind adopting a habit of adding a changelog to all projects (monorepo or not), this is a good enough excuse for that.

**Comment by chriszs** [comment]
[Another consideration.](https://github.com/tvkitchen/appliances/issues/17) But you can still overwrite templates locally even if they have central definition.

**Comment by chriszs** [comment]
Okay, so here's the situation: I started work on a separate PR that added the CoC and CONTRIBUTING.md files to this repo as per #3. That's a little more involved because CONTRIBUTING in particular includes a bunch of relative references to other Markdown files and sections that might be specific to a repo, like Sequelize migration instructions. It's a little unclear how those references are going to work in GitHub and it might involve changing them. The plan was to mark some of those up and work through how to resolve them. But I accidentally committed those to this branch instead of a new one.

Before I could fix it, someone turned off my access to push branches to this repo. So I'm going to pretend I always meant to do that in this PR also.

This has some utility, because the tvkitchen repo lacks a bug template, website and meta lack docs (see this issue in [website](https://github.com/tvkitchen/tv-kitchen-website/issues/1)). Merging this would fix that globally.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Contribution Guidelines
+
+This document contains information of use to developers looking to improve this codebase.  See [README.md](https://github.com/tvkitchen/tv-kitchen/blob/master/README.md) for an introduction to this project. All discussions about this project fall under our [code of conduct](CODE_OF_CONDUCT.md).
```
Comment (chriszs) [inline CONTRIBUTING.md:3]: Changed this to refer to the project, link to the README in the tvkitchen repo.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+## Submitting and Reviewing Code
+
+This project has a home on [GitHub](https://github.com/tvkitchen).  Please submit [pull requests](https://help.github.com/articles/about-pull-requests/) (PRs) there.
+
+Please submit changes via pull request, even if you have direct commit access to the repository.  The PR process allows us to get additional eyes on change proposals, and ensures that your changed code [builds cleanly via Travis CI's automated builds](https://travis-ci.org/tvkitchen/tv-kitchen/).
+
+As you work on your branch, try to test it locally to ensure that it still builds and deploys properly, and that you haven't introduced new accessibility bugs. See `docs/TESTING.md` for more instructions.
```
Comment (chriszs) [inline CONTRIBUTING.md:11]: May need to either put TESTING.md in the same repo or change this so it's no longer a link. Unclear whether GitHub has some magic way of handling this. Should check.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Generally, the more controversial, complex or large a change, the more opportunity people should have to comment on it.  That means it should garner more comments/approvals, or it means it should sit longer before being merged. You can talk with us about a change you'd like to make before or while you work on it; see [the participation section of the README](README.md#participating). We don't have hard rules about such things, and documentation changes usually don't need to sit as long as functional changes, but figure a business day or two for an average patch to get discussed.
```
Comment (chriszs) [inline CONTRIBUTING.md:13]: Another relative link.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+As to when to merge, that's a judgment call.  Usually once an "approved" review goes through, and there aren't any more changes requested, then the author of the PR will merge it (if they have access to push to master).  Generally, wait to merge until the conversation around a change has concluded.  If you're unsure, ask!  "Is this ready to merge?" is often a useful next step in the conversation.
+
+If your PR fixes a bug or adds a feature, please write a test to go with the change (see [TESTING.md](docs/TESTING.md) for details on our testing framework).  If the change involves libraries or would be difficult to test, please use a Given-When-Then description or describe in your PR message how reviewers should test that your change works as expected.
```
Comment (chriszs) [inline CONTRIBUTING.md:17]: Same.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+### Document as you go
+
+If you introduce a new framework or dependency, add the necessary information to [INSTALL.md](docs/INSTALL.md) and the other documentation.
```
Comment (chriszs) [inline CONTRIBUTING.md:21]: Another relative link.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+### Configurations
+
+We're using `dotenv` for general runtime configuration settings. Never hard code values that should be configurable, but instead edit the .env.template file.
+
+Commits that introduce a new configurable variable should also document the configuration steps in the commit message.
+
+### Migrations
+
+The golden rule of migrations: Never edit a migration file that has been committed to master.
```
Comment (chriszs) [inline CONTRIBUTING.md:31]: Are Sequelize migration instructions applicable globally?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+		# Delete it upstream.
+		$ git push origin --delete some-now-fully-merged-branch
+
+## Avoiding Generatable Elements In The Repo
+
+We try to keep generated elements out of the repository.  This includes files that result from build processes.  If we want to memorialize a compiled version of the program, the best way to do that is with tags or to record that information and put the saved version somewhere other than this repository.  If a file can be generated from the materials in the repository using commonly-available tools, please do not put it in the repository without raising it for discussion.
+
+## Thank you!
+
+The [Corporate Overlords](https://biffud.com/people) of [Bad Idea Factory](https://biffud.com/) appreciate your dedication and service to the public good.
```
Comment (chriszs) [inline CONTRIBUTING.md:171]: Added some friendly, explanatory links to BIF's website here.

**Comment by reefdog** [comment]
FWIW I think it's perfectly fine to remove the links if we can't wrestle them into relative submission. It's Very Not Hard for someone to just look up `TESTING.md` wherever it is in the repo.

**Comment by chriszs** [comment]
Makes sense to me.

**Comment by slifty** [comment]
Wait how did you lose access to push branches (I did not change anything!)

File: CONTRIBUTING.md (hunk 7 above)
Comment (slifty) [reply CONTRIBUTING.md:31]: I think we could keep this but make it more general -- "For projects with migrations...."

**Comment by slifty** [comment]
I agree with both of you (I think you both agree, right?) -- and we can always add links back in later after shifting over to this model.

**Review by slifty — APPROVED** [review]

**Comment by slifty** [comment]
Gonna merge these in, pray for mojo...

**Pull request merged**